src/python/pants/backend/docker/rules_test.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from hashlib import sha256
+from os import path
+from textwrap import dedent
+
+from pants.backend.docker.rules import (
+    DockerBinary,
+    InjectDockerDependencies,
+    inject_docker_dependencies,
+    parse_dockerfile,
+)
+from pants.backend.docker.target_types import DockerDependencies, DockerImage
+from pants.backend.python.target_types import PexBinary
+from pants.engine.addresses import Address
+from pants.engine.fs import Digest
+from pants.engine.process import Process
+from pants.engine.target import InjectedDependencies
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+
+
```

Comment:
would be nice with at least one integration like test in here, testing the `build_docker_image` rule.. but I'm a bit unsure how to write one.. There's stuff that would need to be mocked, at least :)